components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenRequestBuilderTest.java

```diff
@@ -0,0 +1,417 @@
+            Assert.assertEquals(expectedOperation.getPaths().size(), actualOperation.getPaths().size());
+            for (int j = 0; j < expectedOperation.getPaths().size(); j++) {
+                Assert.assertEquals(expectedOperation.getPaths().get(j), actualOperation.getPaths().get(j));
+            }
+        }
+    }
+
+    /**
+     * Mock the token message context for testing.
+     *
+     * @return A map representing the event context.
+     */
+    private Map<String, Object> mockTokenMessageContext() {
+
+        Map<String, Object> eventContext = new HashMap<>();
+
+        OAuth2AccessTokenReqDTO tokenReqDTO = mockTokenRequestDTO();
+        AuthenticatedUser authenticatedUser = mockAuthenticatedUser();
+        OAuthTokenReqMessageContext tokenMessageContext = mockMessageContext(tokenReqDTO, authenticatedUser);
+        eventContext.put("tokenMessageContext", tokenMessageContext);
+
+        return eventContext;
+    }
+
+    /**
+     * Mock the OAuth2 access token request DTO.
+     *
+     * @return OAuth2AccessTokenReqDTO containing mock token request data.
+     */
+    private OAuth2AccessTokenReqDTO mockTokenRequestDTO() {
+
+        OAuth2AccessTokenReqDTO tokenReqDTO = new OAuth2AccessTokenReqDTO();
+        tokenReqDTO.setClientId(CLIENT_ID_TEST);
+        tokenReqDTO.setClientSecret(CLIENT_SECRET_TEST);
+        tokenReqDTO.setGrantType(GRANT_TYPE_TEST);
+        tokenReqDTO.setTenantDomain(TENANT_DOMAIN_TEST);
+        tokenReqDTO.setResourceOwnerUsername(USERNAME_TEST);
+        tokenReqDTO.setResourceOwnerPassword(PASSWORD_TEST);
+        tokenReqDTO.setScope(new String[]{"scope1", "scope2"});
+        HttpRequestHeader[] requestHeaders = new HttpRequestHeader[]{
+                new HttpRequestHeader("authorization",
+                        getBase64EncodedString(CLIENT_ID_TEST, CLIENT_SECRET_TEST)),
+                new HttpRequestHeader("accept", "application/json")
+        };
+        tokenReqDTO.setHttpRequestHeaders(requestHeaders);
+        RequestParameter[] requestParameters = new RequestParameter[] {
+                new RequestParameter("grant_type", GRANT_TYPE_TEST),
+                new RequestParameter("username", USERNAME_TEST),
+                new RequestParameter("password", PASSWORD_TEST),
+                new RequestParameter("scope", "scope1", "scope2")
+        };
+        tokenReqDTO.setRequestParameters(requestParameters);
+        return tokenReqDTO;
+    }
+
+    /**
+     * Mock an authenticated user.
+     *
+     * @return AuthenticatedUser object containing mock user data.
+     */
+    private static AuthenticatedUser mockAuthenticatedUser() {
+
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        authenticatedUser.setUserName(USERNAME_TEST);
+        authenticatedUser.setUserStoreDomain(USER_STORE_TEST);
+        authenticatedUser.setUserId(USER_ID_TEST);
+        authenticatedUser.setAuthenticatedSubjectIdentifier(CLIENT_ID_TEST);
```

Comment:
Let's use the username for this one as well

Reply:
This is updated with ef43ee44e99c969f75d05fe1641cb112f816a7a4